core/shared/src/main/scala/scalaz/zio/Duration.scala

```diff
@@ -1,25 +1,29 @@
 package scalaz.zio
 
-sealed trait Duration {
-  def toDuration: scala.concurrent.duration.Duration
-}
+sealed trait Duration
 
 object Duration {
 
-  case class Finite(duration: scala.concurrent.duration.FiniteDuration) extends Duration {
-    final def toDuration = duration
-  }
+  class Finite(duration: scala.concurrent.duration.FiniteDuration) extends Duration {
+    
+    def toMillis: Long = duration.toMillis
+
+    def toNanos: Long = duration.toNanos
 
-  case class Infinite() extends Duration {
-    final def toDuration = scala.concurrent.duration.Duration.Inf
+    def *(factor: Double): Duration = Duration.fromNanos((this.toNanos * factor).round)
   }
 
-  def fromNanos(length: Long): Finite = Finite(scala.concurrent.duration.Duration.fromNanos(length))
+  case class Infinite() extends Duration
 
-  def fromDuration(duration: scala.concurrent.duration.Duration): Either[IllegalArgumentException, Duration] =
+  def fromNanos(nanos: Long): Finite = new Finite(scala.concurrent.duration.Duration.fromNanos(nanos))
+
+  def fromDuration(duration: scala.concurrent.duration.Duration): Option[Duration] =
     duration match {
-      case finiteDuration: scala.concurrent.duration.FiniteDuration => Right(Finite(finiteDuration))
-      case inf if inf == scala.concurrent.duration.Duration.Inf     => Right(Infinite())
-      case x                                                        => Left(new IllegalArgumentException(s"Woah dude, $x is not a meaningful zio.Duration!"))
+      case finiteDuration: scala.concurrent.duration.FiniteDuration => Some(new Finite(finiteDuration))
+      case inf if inf == scala.concurrent.duration.Duration.Inf     => Some(Infinite())
+      case _                                                        => None
     }
+
+  val Zero: Finite = fromNanos(0)
```

Comment:
This should return `Duration`, not `Finite`.

Reply:
But Zero is Finite. Isn't precise type better here?